### ISSUE TITLE
Error: It's not allowed to call mmap on 0x400000

### ISSUE THREAD

**Issue by neokril** [tsgates/mbox#10]
I keep getting this error if run any program under mbox without parameters.

```
$ ./mbox  ls

Stop executing pid=15305: It's not allowed to call mmap on 0x400000
Sandbox Root:
 > /tmp/sandbox-15301
```

However it works without any problems with `-s`. 

Is it an expected behavior?

_Environment: Ubuntu 12.04.1 LTS x64 run inside VirtualBox 4.2.6_

**Comment by tsgates** [comment]
mbox looks up code section in elf binaries to place trampoline call/args, but your binary itself attempts to access that memory region (around entry point of binary), where adversaries can possibly trick mbox not to rewrite system call arguments. 

by specifying -s, mbox might choose different memory region for trampoline. I will fix this issue on ubuntu, whenever I can find some free time.

**Comment by joshbaptiste** [comment]
*Arch
snoopy:~ $ mbox -sn -- wget google.com ; uname -a
Stop executing pid=1929: It's not allowed to call mmap on 0x400000
Sandbox Root:

> /tmp/sandbox-1925
> Linux snoopy 3.8.6-1-ARCH #1 SMP PREEMPT Sat Apr 6 07:27:01 CEST 2013 x86_64 GNU/Linux

*Fedora 18
miagos:~ $ mbox -sn -- wget google.com ; uname -a
Stop executing pid=28530: It's not allowed to call mmap on 0x400000
Sandbox Root:

> /tmp/sandbox-28526
> Linux miagos 3.9.6-200.fc18.x86_64 #1 SMP Thu Jun 13 18:56:55 UTC 2013 x86_64 x86_64 x86_64 GNU/Linux

*Mint (aka Ubuntu)
linus:~ $ mbox -sn -- wget google.com ; uname -a
mbox: exec: Operation not permitted
Sandbox Root:

> /tmp/sandbox-30247
> Linux linus 3.5.0-26-generic #42-Ubuntu SMP Fri Mar 8 23:18:20 UTC 2013 x86_64 x86_64 x86_64 GNU/Linux

**Comment by heinrich5991** [comment]
Also getting this for the more interesting tools.

What works:
- `bash`
- `gcc`
- `gdb`!
- `ls`
- `sh`
- `mkdir`
- `rm`
- `uname`
- `vim`

What doesn't:
- `emacs`
- `mbox` (that's really sad!)
- `pacman` (archlinux package manager)
- `python`
- `wget`

`-s` doesn't seem to fix those issues.

**Comment by tsgates** [comment]
Could you give me an favour? If 'ls' failed, could you give me a trace of 'strace ls' to check why ls mmaps on 0x400000, where is the entry point of the binary?

**Comment by heinrich5991** [comment]
```
$ strace  wget -o /dev/null foo
execve("/usr/bin/wget", ["wget", "-o", "/dev/null", "foo"], [/* 33 vars */]) = 0
brk(0)                                  = 0x2319000
access("/etc/ld.so.preload", R_OK)      = -1 ENOENT (No such file or directory)
open("/etc/ld.so.cache", O_RDONLY|O_CLOEXEC) = 3
fstat(3, {st_mode=S_IFREG|0644, st_size=249822, ...}) = 0
mmap(NULL, 249822, PROT_READ, MAP_PRIVATE, 3, 0) = 0x7fe4d865f000
close(3)                                = 0
open("/usr/lib/libssl.so.1.0.0", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\240y\1\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0555, st_size=492396, ...}) = 0
mmap(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7fe4d865e000
mmap(NULL, 2539824, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7fe4d820f000
mprotect(0x7fe4d8272000, 2093056, PROT_NONE) = 0
mmap(0x7fe4d8471000, 45056, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x62000) = 0x7fe4d8471000
close(3)                                = 0
open("/usr/lib/libcrypto.so.1.0.0", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\300,\6\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0555, st_size=2378755, ...}) = 0
mmap(NULL, 4218072, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7fe4d7e09000
mprotect(0x7fe4d7fe6000, 2093056, PROT_NONE) = 0
mmap(0x7fe4d81e5000, 155648, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x1dc000) = 0x7fe4d81e5000
mmap(0x7fe4d820b000, 15576, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_ANONYMOUS, -1, 0) = 0x7fe4d820b000
close(3)                                = 0
open("/usr/lib/libz.so.1", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0000#\0\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0755, st_size=88592, ...}) = 0
mmap(NULL, 2183688, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7fe4d7bf3000
mprotect(0x7fe4d7c08000, 2093056, PROT_NONE) = 0
mmap(0x7fe4d7e07000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x14000) = 0x7fe4d7e07000
close(3)                                = 0
open("/usr/lib/libidn.so.11", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0`1\0\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0755, st_size=207336, ...}) = 0
mmap(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7fe4d865d000
mmap(NULL, 2302432, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7fe4d79c0000
mprotect(0x7fe4d79f2000, 2093056, PROT_NONE) = 0
mmap(0x7fe4d7bf1000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x31000) = 0x7fe4d7bf1000
close(3)                                = 0
open("/usr/lib/libuuid.so.1", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0P\25\0\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0755, st_size=19016, ...}) = 0
mmap(NULL, 2113928, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7fe4d77bb000
mprotect(0x7fe4d77bf000, 2093056, PROT_NONE) = 0
mmap(0x7fe4d79be000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x3000) = 0x7fe4d79be000
close(3)                                = 0
open("/usr/lib/libpcre.so.1", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\20\27\0\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0755, st_size=420096, ...}) = 0
mmap(NULL, 2515496, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7fe4d7554000
mprotect(0x7fe4d75ba000, 2093056, PROT_NONE) = 0
mmap(0x7fe4d77b9000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x65000) = 0x7fe4d77b9000
close(3)                                = 0
open("/usr/lib/libc.so.6", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\3\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\20\34\2\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0755, st_size=2022349, ...}) = 0
mmap(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7fe4d865c000
mmap(NULL, 3832336, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7fe4d71ac000
mprotect(0x7fe4d734a000, 2097152, PROT_NONE) = 0
mmap(0x7fe4d754a000, 24576, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x19e000) = 0x7fe4d754a000
mmap(0x7fe4d7550000, 14864, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_ANONYMOUS, -1, 0) = 0x7fe4d7550000
close(3)                                = 0
open("/usr/lib/libdl.so.2", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\320\16\0\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0755, st_size=14672, ...}) = 0
mmap(NULL, 2109712, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7fe4d6fa8000
mprotect(0x7fe4d6fab000, 2093056, PROT_NONE) = 0
mmap(0x7fe4d71aa000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x2000) = 0x7fe4d71aa000
close(3)                                = 0
open("/usr/lib/libpthread.so.0", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\20o\0\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0755, st_size=144995, ...}) = 0
mmap(NULL, 2212976, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7fe4d6d8b000
mprotect(0x7fe4d6da3000, 2093056, PROT_NONE) = 0
mmap(0x7fe4d6fa2000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x17000) = 0x7fe4d6fa2000
mmap(0x7fe4d6fa4000, 13424, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_ANONYMOUS, -1, 0) = 0x7fe4d6fa4000
close(3)                                = 0
mmap(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7fe4d865b000
mmap(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7fe4d865a000
mmap(NULL, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7fe4d8658000
arch_prctl(ARCH_SET_FS, 0x7fe4d8658780) = 0
mprotect(0x7fe4d754a000, 16384, PROT_READ) = 0
mprotect(0x7fe4d6fa2000, 4096, PROT_READ) = 0
mprotect(0x7fe4d71aa000, 4096, PROT_READ) = 0
mprotect(0x7fe4d77b9000, 4096, PROT_READ) = 0
mprotect(0x7fe4d79be000, 4096, PROT_READ) = 0
mprotect(0x7fe4d7bf1000, 4096, PROT_READ) = 0
mprotect(0x7fe4d7e07000, 4096, PROT_READ) = 0
mprotect(0x7fe4d81e5000, 110592, PROT_READ) = 0
mprotect(0x7fe4d8471000, 16384, PROT_READ) = 0
mprotect(0x65f000, 4096, PROT_READ)     = 0
mprotect(0x7fe4d869c000, 4096, PROT_READ) = 0
munmap(0x7fe4d865f000, 249822)          = 0
set_tid_address(0x7fe4d8658a50)         = 13830
set_robust_list(0x7fe4d8658a60, 24)     = 0
rt_sigaction(SIGRTMIN, {0x7fe4d6d919f0, [], SA_RESTORER|SA_SIGINFO, 0x7fe4d6d9a880}, NULL, 8) = 0
rt_sigaction(SIGRT_1, {0x7fe4d6d91a80, [], SA_RESTORER|SA_RESTART|SA_SIGINFO, 0x7fe4d6d9a880}, NULL, 8) = 0
rt_sigprocmask(SIG_UNBLOCK, [RTMIN RT_1], NULL, 8) = 0
getrlimit(RLIMIT_STACK, {rlim_cur=8192*1024, rlim_max=RLIM64_INFINITY}) = 0
brk(0)                                  = 0x2319000
brk(0x233a000)                          = 0x233a000
clock_getres(CLOCK_MONOTONIC, {0, 1})   = 0
clock_getres(CLOCK_MONOTONIC, {0, 1})   = 0
open("/usr/lib/locale/locale-archive", O_RDONLY|O_CLOEXEC) = 3
fstat(3, {st_mode=S_IFREG|0644, st_size=3116592, ...}) = 0
mmap(NULL, 3116592, PROT_READ, MAP_PRIVATE, 3, 0) = 0x7fe4d6a92000
close(3)                                = 0
stat("/etc/wgetrc", {st_mode=S_IFREG|0644, st_size=4896, ...}) = 0
open("/etc/wgetrc", O_RDONLY)           = 3
fstat(3, {st_mode=S_IFREG|0644, st_size=4896, ...}) = 0
mmap(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7fe4d869b000
read(3, "###\n### Sample Wget initializati"..., 4096) = 4096
read(3, "s) by setting this to on.\n#recur"..., 4096) = 800
read(3, "", 4096)                       = 0
close(3)                                = 0
munmap(0x7fe4d869b000, 4096)            = 0
stat("/home/hein/.wgetrc", 0x7fff5a556320) = -1 ENOENT (No such file or directory)
open("/dev/null", O_WRONLY|O_CREAT|O_TRUNC, 0666) = 3
rt_sigaction(SIGHUP, {SIG_IGN, [HUP], SA_RESTORER|SA_RESTART, 0x7fe4d71e1400}, {SIG_DFL, [], 0}, 8) = 0
rt_sigaction(SIGHUP, {0x421460, [HUP], SA_RESTORER|SA_RESTART, 0x7fe4d71e1400}, {SIG_IGN, [HUP], SA_RESTORER|SA_RESTART, 0x7fe4d71e1400}, 8) = 0
rt_sigaction(SIGUSR1, {0x421460, [USR1], SA_RESTORER|SA_RESTART, 0x7fe4d71e1400}, {SIG_DFL, [], 0}, 8) = 0
rt_sigaction(SIGPIPE, {SIG_IGN, [PIPE], SA_RESTORER|SA_RESTART, 0x7fe4d71e1400}, {SIG_DFL, [], 0}, 8) = 0
rt_sigaction(SIGWINCH, {0x422b90, [WINCH], SA_RESTORER|SA_RESTART, 0x7fe4d71e1400}, {SIG_DFL, [], 0}, 8) = 0
statfs(".", {f_type="EXT2_SUPER_MAGIC", f_bsize=4096, f_blocks=98398167, f_bfree=14320517, f_bavail=9316417, f_files=25001984, f_ffree=23540871, f_fsid={-1176054900, 723446372}, f_namelen=255, f_frsize=4096}) = 0
stat("index.html", 0x7fff5a555f00)      = -1 ENOENT (No such file or directory)
stat("index.html", 0x7fff5a555fa0)      = -1 ENOENT (No such file or directory)
statfs(".", {f_type="EXT2_SUPER_MAGIC", f_bsize=4096, f_blocks=98398167, f_bfree=14320517, f_bavail=9316417, f_files=25001984, f_ffree=23540871, f_fsid={-1176054900, 723446372}, f_namelen=255, f_frsize=4096}) = 0
stat("index.html", 0x7fff5a555f00)      = -1 ENOENT (No such file or directory)
open("/etc/localtime", O_RDONLY|O_CLOEXEC) = 4
fstat(4, {st_mode=S_IFREG|0644, st_size=2309, ...}) = 0
fstat(4, {st_mode=S_IFREG|0644, st_size=2309, ...}) = 0
mmap(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7fe4d869b000
read(4, "TZif2\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\10\0\0\0\10\0\0\0\0"..., 4096) = 2309
lseek(4, -1467, SEEK_CUR)               = 842
read(4, "TZif2\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\t\0\0\0\t\0\0\0\0"..., 4096) = 1467
close(4)                                = 0
munmap(0x7fe4d869b000, 4096)            = 0
fstat(3, {st_mode=S_IFCHR|0666, st_rdev=makedev(1, 3), ...}) = 0
ioctl(3, SNDCTL_TMR_TIMEBASE or SNDRV_TIMER_IOCTL_NEXT_DEVICE or TCGETS, 0x7fff5a555700) = -1 ENOTTY (Inappropriate ioctl for device)
mmap(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7fe4d869b000
write(3, "--2014-02-14 16:28:54--  http://"..., 37) = 37
stat("/home/hein/.netrc", 0x7fff5a555a00) = -1 ENOENT (No such file or directory)
write(3, "Resolving foo (foo)... ", 23) = 23
socket(PF_LOCAL, SOCK_STREAM|SOCK_CLOEXEC|SOCK_NONBLOCK, 0) = 4
connect(4, {sa_family=AF_LOCAL, sun_path="/var/run/nscd/socket"}, 110) = -1 ENOENT (No such file or directory)
close(4)                                = 0
socket(PF_LOCAL, SOCK_STREAM|SOCK_CLOEXEC|SOCK_NONBLOCK, 0) = 4
connect(4, {sa_family=AF_LOCAL, sun_path="/var/run/nscd/socket"}, 110) = -1 ENOENT (No such file or directory)
close(4)                                = 0
open("/etc/nsswitch.conf", O_RDONLY|O_CLOEXEC) = 4
fstat(4, {st_mode=S_IFREG|0644, st_size=234, ...}) = 0
mmap(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7fe4d869a000
read(4, "# Begin /etc/nsswitch.conf\n\npass"..., 4096) = 234
read(4, "", 4096)                       = 0
close(4)                                = 0
munmap(0x7fe4d869a000, 4096)            = 0
open("/etc/host.conf", O_RDONLY|O_CLOEXEC) = 4
fstat(4, {st_mode=S_IFREG|0644, st_size=63, ...}) = 0
mmap(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7fe4d869a000
read(4, "#\n# /etc/host.conf\n#\n\norder host"..., 4096) = 63
read(4, "", 4096)                       = 0
close(4)                                = 0
munmap(0x7fe4d869a000, 4096)            = 0
futex(0x7fe4d7552be4, FUTEX_WAKE_PRIVATE, 2147483647) = 0
open("/etc/resolv.conf", O_RDONLY|O_CLOEXEC) = 4
fstat(4, {st_mode=S_IFREG|0644, st_size=119, ...}) = 0
mmap(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7fe4d869a000
read(4, "#\n# /etc/resolv.conf\n#\n\n#search "..., 4096) = 119
read(4, "", 4096)                       = 0
close(4)                                = 0
munmap(0x7fe4d869a000, 4096)            = 0
uname({sys="Linux", node="arch5", ...}) = 0
open("/etc/ld.so.cache", O_RDONLY|O_CLOEXEC) = 4
fstat(4, {st_mode=S_IFREG|0644, st_size=249822, ...}) = 0
mmap(NULL, 249822, PROT_READ, MAP_PRIVATE, 4, 0) = 0x7fe4d861b000
close(4)                                = 0
open("/usr/lib/libnss_files.so.2", O_RDONLY|O_CLOEXEC) = 4
read(4, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\220\"\0\0\0\0\0\0"..., 832) = 832
fstat(4, {st_mode=S_IFREG|0755, st_size=47712, ...}) = 0
mmap(NULL, 2144392, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 4, 0) = 0x7fe4d6886000
mprotect(0x7fe4d6891000, 2093056, PROT_NONE) = 0
mmap(0x7fe4d6a90000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 4, 0xa000) = 0x7fe4d6a90000
close(4)                                = 0
mprotect(0x7fe4d6a90000, 4096, PROT_READ) = 0
munmap(0x7fe4d861b000, 249822)          = 0
open("/etc/hosts", O_RDONLY|O_CLOEXEC)  = 4
fstat(4, {st_mode=S_IFREG|0644, st_size=270, ...}) = 0
mmap(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7fe4d869a000
read(4, "#\n# /etc/hosts: static lookup ta"..., 4096) = 270
read(4, "", 4096)                       = 0
close(4)                                = 0
munmap(0x7fe4d869a000, 4096)            = 0
open("/etc/ld.so.cache", O_RDONLY|O_CLOEXEC) = 4
fstat(4, {st_mode=S_IFREG|0644, st_size=249822, ...}) = 0
mmap(NULL, 249822, PROT_READ, MAP_PRIVATE, 4, 0) = 0x7fe4d861b000
close(4)                                = 0
open("/usr/lib/libnss_dns.so.2", O_RDONLY|O_CLOEXEC) = 4
read(4, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\0\21\0\0\0\0\0\0"..., 832) = 832
fstat(4, {st_mode=S_IFREG|0755, st_size=22960, ...}) = 0
mmap(NULL, 2117896, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 4, 0) = 0x7fe4d6680000
mprotect(0x7fe4d6685000, 2093056, PROT_NONE) = 0
mmap(0x7fe4d6884000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 4, 0x4000) = 0x7fe4d6884000
close(4)                                = 0
open("/usr/lib/libresolv.so.2", O_RDONLY|O_CLOEXEC) = 4
read(4, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\220:\0\0\0\0\0\0"..., 832) = 832
fstat(4, {st_mode=S_IFREG|0755, st_size=84864, ...}) = 0
mmap(NULL, 2189928, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 4, 0) = 0x7fe4d6469000
mprotect(0x7fe4d647d000, 2093056, PROT_NONE) = 0
mmap(0x7fe4d667c000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 4, 0x13000) = 0x7fe4d667c000
mmap(0x7fe4d667e000, 6760, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_ANONYMOUS, -1, 0) = 0x7fe4d667e000
close(4)                                = 0
mprotect(0x7fe4d667c000, 4096, PROT_READ) = 0
mprotect(0x7fe4d6884000, 4096, PROT_READ) = 0
munmap(0x7fe4d861b000, 249822)          = 0
socket(PF_INET, SOCK_DGRAM|SOCK_NONBLOCK, IPPROTO_IP) = 4
connect(4, {sa_family=AF_INET, sin_port=htons(53), sin_addr=inet_addr("8.8.8.8")}, 16) = 0
poll([{fd=4, events=POLLOUT}], 1, 0)    = 1 ([{fd=4, revents=POLLOUT}])
sendmmsg(4, {{{msg_name(0)=NULL, msg_iov(1)=[{"\352i\1\0\0\1\0\0\0\0\0\0\3foo\0\0\1\0\1", 21}], msg_controllen=0, msg_flags=0}, 21}, {{msg_name(0)=NULL, msg_iov(1)=[{"\30\325\1\0\0\1\0\0\0\0\0\0\3foo\0\0\34\0\1", 21}], msg_controllen=0, msg_flags=0}, 21}}, 2, MSG_NOSIGNAL) = 2
poll([{fd=4, events=POLLIN}], 1, 5000)  = 1 ([{fd=4, revents=POLLIN}])
ioctl(4, FIONREAD, [96])                = 0
recvfrom(4, "\352i\201\203\0\1\0\0\0\1\0\0\3foo\0\0\1\0\1\0\0\6\0\1\0\0\6\344\0@"..., 2048, 0, {sa_family=AF_INET, sin_port=htons(53), sin_addr=inet_addr("8.8.8.8")}, [16]) = 96
poll([{fd=4, events=POLLIN}], 1, 4953)  = 1 ([{fd=4, revents=POLLIN}])
ioctl(4, FIONREAD, [96])                = 0
recvfrom(4, "\30\325\201\203\0\1\0\0\0\1\0\0\3foo\0\0\34\0\1\0\0\6\0\1\0\0\5,\0@"..., 1952, 0, {sa_family=AF_INET, sin_port=htons(53), sin_addr=inet_addr("8.8.8.8")}, [16]) = 96
close(4)                                = 0
open("/etc/ld.so.cache", O_RDONLY|O_CLOEXEC) = 4
fstat(4, {st_mode=S_IFREG|0644, st_size=249822, ...}) = 0
mmap(NULL, 249822, PROT_READ, MAP_PRIVATE, 4, 0) = 0x7fe4d861b000
close(4)                                = 0
open("/usr/lib/libnss_myhostname.so.2", O_RDONLY|O_CLOEXEC) = 4
read(4, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\0\r\0\0\0\0\0\0"..., 832) = 832
fstat(4, {st_mode=S_IFREG|0755, st_size=14272, ...}) = 0
mmap(NULL, 2109456, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 4, 0) = 0x7fe4d6265000
mprotect(0x7fe4d6268000, 2093056, PROT_NONE) = 0
mmap(0x7fe4d6467000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 4, 0x2000) = 0x7fe4d6467000
close(4)                                = 0
mprotect(0x7fe4d6467000, 4096, PROT_READ) = 0
munmap(0x7fe4d861b000, 249822)          = 0
uname({sys="Linux", node="arch5", ...}) = 0
write(3, "failed: Name or service not know"..., 35) = 35
open("/usr/lib/charset.alias", O_RDONLY|O_NOFOLLOW) = -1 ENOENT (No such file or directory)
write(3, "wget: unable to resolve host add"..., 47) = 47
stat("/etc/localtime", {st_mode=S_IFREG|0644, st_size=2309, ...}) = 0
close(3)                                = 0
munmap(0x7fe4d869b000, 4096)            = 0
exit_group(4)                           = ?
+++ exited with 4 +++
```

**Comment by tsgates** [comment]
thanks! could you check it out for me again?

**Comment by hiciu** [comment]
5fe59e1 fixed this for me (I'm actually testing with 3cafe2f0...):

```
$ mbox ls              
dist  doc  Makefile  README.md  src
Sandbox Root:
 > /tmp/sandbox-5670
$ mbox python
Python 3.3.3 (default, Nov 26 2013, 13:33:18) 
[GCC 4.8.2] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> 
Sandbox Root:
 > /tmp/sandbox-5704
$ mbox emacs 
mbox: Can't stat 'emacs': No such file or directory
$ mbox wget -O /dev/null foo
--2014-02-14 22:36:43--  http://foo/
Translacja foo (foo)... nieudane: Ta nazwa lub usługa jest nieznana.
wget: nie udało się rozwiązać adresu hosta `foo'
Sandbox Root:
 > /tmp/sandbox-6011
Network Summary:
 > [6015] -> 127.0.0.1:53
 > [6015] Create socket(PF_INET,...)
```

With pacman I'm getting a lot of `process_vm_readv: Operation not permitted`, but otherwise it works fine (I'm pretty sure it is separate issue, not related to this one).

Thank you!

**Comment by heinrich5991** [comment]
This fixes it for me, thanks!
(even `pacman` works!)
**EDIT**: `pacman` doesn't work, same error as @hiciu
